src/annotation.js

```diff
@@ -0,0 +1,607 @@
+var $ = require('jquery');
+var inherit = require('./inherit');
+var geo_event = require('./event');
+var transform = require('./transform');
+
+var annotationId = 0;
+
+var annotationState = {
+  create: 'create',
+  done: 'done',
+  edit: 'edit'
+};
+
+/////////////////////////////////////////////////////////////////////////////
+/**
+ * Base annotation class
+ *
+ * @class geo.annotation
+ * @param {string} type the type of annotation.
+ * @param {object?} options Inidividual annotations have additional options.
+ * @param {string} [options.name] A name for the annotation.  This defaults to
+ *    the type with a unique ID suffixed to it.
+ * @param {geo.annotationLayer} [options.layer] a reference to the controlling
+ *    layer.  This is used for coordinate transforms.
+ * @param {string} [options.state] initial annotation state.  One of the
+ *    annotation.state values.
+ * @returns {geo.annotation}
+ */
+/////////////////////////////////////////////////////////////////////////////
+var annotation = function (type, args) {
```

Comment:
Each annotation subclass defines the type, so this is fully extensible and types aren't predefined.  I don't expect users to call the annotation base class by itself unless they are defining a new type.  Internally, we have `polygon`, `rectangle`, and `point` right now, but I could see adding lots more -- lines, labels, ellipses, etc.